### ISSUE TITLE
auto-ROI crashed on sample  HMV_A2A-28-29-30-31-32-33

### ISSUE THREAD

**Issue by raacampbell** [SainsburyWellcomeCentre/BakingTray#398]
This is a 6 brain 2.2 x 2.2 x 5 micron acquisition. This hasn't been tried before as far as I know but I don't see how the following happened or why. 

```
BAKE FAILED IN acquisition_view. CAUGHT THE FOLLOWING ERROR:
 Index exceeds the number of array elements (125).
Line 63 in C:\MATLAB\BakingTray\code\autoROI\+autoROI\+autothresh\run.m
Line 411 in C:\MATLAB\BakingTray\code\autoROI\autoROI.m
Line 132 in C:\MATLAB\BakingTray\code\@BT\getNextROIs.m
Line 332 in C:\MATLAB\BakingTray\code\@BT\bake.m
Line 52 in C:\MATLAB\BakingTray\code\+BakingTray\+gui\@acquisition_view\bake_callback.m
Line 61 in C:\MATLAB\BakingTray\code\+BakingTray\+gui\@acquisition_view\buildFigure.m

Index exceeds the number of array elements (125).

Error in autoROI.autothresh.run (line 63)
        BB = BBstats.roiStats(pStack.sectionNumber).BoundingBoxes;

Error in autoROI (line 411)
            [tThreshSD,~,thresh]=autoROI.autothresh.run(pStack,[],[],out);

Error in BT/getNextROIs (line 132)
    tStats = autoROI(pStack, ...

Error in BT/bake (line 332)
            success = obj.getNextROIs;

Error in BakingTray.gui.acquisition_view/bake_callback (line 52)
        sectionInd = obj.model.bake; %if the bake loop didn't start, it returns 0

Error in BakingTray.gui.acquisition_view/buildFigure>@(varargin)obj.bake_callback(varargin{:}) (line 61)
                    'Callback', @obj.bake_callback, ...
 
Error while evaluating UIControl Callback.
```

**Comment by raacampbell** [comment]
I think the same thing happened again on CBS.

```

After ROI shift we have:
#1 Corner position: 0/1  Box Size: 463/338

Running autoROI with provided threshold of -7431.06
Expanding ROI due to sample clipping!

TRIGGERING RE-CALC OF tThreshSD due to high F/B ratio.



In autoThresh.run

 ***** WARNING PSTACK SLICES: 1. CURRENT SECTION NUMBER: 161
Likely re-calculating thresh in live acq. Forcing sectionNumber to equal stack length
  MSG#1 -- DETACH LOGGER -- [loghandler,loghandler.detachLogObject -> bakeCleanupFun -> @()bakeCleanupFun(obj) -> onCleanup.delete -> bake -> bake_callback -> @(varargin)obj.bake_callback(varargin{:})]

Closed log files

Disarmed scanner: 08-Sep-2021 03:07:13
resetting abort flags
Tearing down scanner
Requested 243 slices: this is 9.72 mm thick but only 3.95 mm is possible. Setting to the maximum value of 97 slices.

BAKE FAILED IN acquisition_view. CAUGHT THE FOLLOWING ERROR:
 Index exceeds the number of array elements (148).
Line 63 in C:\MATLAB\BakingTray\code\autoROI\+autoROI\+autothresh\run.m
Line 411 in C:\MATLAB\BakingTray\code\autoROI\autoROI.m
Line 132 in C:\MATLAB\BakingTray\code\@BT\getNextROIs.m
Line 332 in C:\MATLAB\BakingTray\code\@BT\bake.m
Line 61 in C:\MATLAB\BakingTray\code\+BakingTray\+gui\@acquisition_view\bake_callback.m
Line 61 in C:\MATLAB\BakingTray\code\+BakingTray\+gui\@acquisition_view\buildFigure.m

Index exceeds the number of array elements (148).
Error in autoROI.autothresh.run (line 63)
        BB = BBstats.roiStats(pStack.sectionNumber).BoundingBoxes;
Error in autoROI (line 411)
            [tThreshSD,~,thresh]=autoROI.autothresh.run(pStack,[],[],out);
Error in BT/getNextROIs (line 132)
    tStats = autoROI(pStack, ...
Error in BT/bake (line 332)
            success = obj.getNextROIs;
Error in BakingTray.gui.acquisition_view/bake_callback (line 61)
        sectionInd = obj.model.bake; %if the bake loop didn't start, it returns 0
Error in BakingTray.gui.acquisition_view/buildFigure>@(varargin)obj.bake_callback(varargin{:}) (line 61)
                    'Callback', @obj.bake_callback, ... 
Error while evaluating UIControl Callback.

```

**Comment by raacampbell** [comment]
Ok... so:

```
>> load CBS_AUTO_ROI_FAIL/rawData/auto_ROI_stats.mat
>> autoROI_stats.stats.roiStats

ans = 

  1x147 struct array with fields:
```

And above it says:
```
BAKE FAILED IN acquisition_view. CAUGHT THE FOLLOWING ERROR:
 Index exceeds the number of array elements (148). <--- HERE
Line 63 in C:\MATLAB\BakingTray\code\autoROI\+autoROI\+autothresh\run.m
```

So probably that's the issue. We are trying to access index 148 and that does not exist. It happens during re-triggering of the auto-thresh, which is probably a rare event. Maybe it will fail this way every time. 

Fixed here: 03244e8e4d05079f6b1f034006c7d4e038524273

**Comment by raacampbell** [comment]
This no longer happens, must be fixed.
